public/js/search.js

```diff
@@ -0,0 +1,66 @@
+$(function(config){
+  var algolia = algoliasearch(config.applicationId, config.apiKey);
+  var index = algolia.initIndex(config.indexName);
+
+  hideSearch();
+
+  $('.clear-search img, .search-results-container img').on('click', function() {
+    $('.sidebar-navigation--search input').val('');
+    hideSearch();
+  });
+
+  $('.sidebar-navigation--search input').on('input', function() {
+    var query = $(this).val();
+    onQueryChange(query);
+  });
+
+  function onQueryChange(query) {
+    if (query.length) {
+      index.search(query, {'hitsPerPage': '10', 'page': '0', 'typoTolerance': 'false'}, onResult);
```

Comment:
Does this second argument have to be a JSON object or just a regular JavaScript object? (Totally just curious, I don't usually see js object properties with quotes around them but I know nothing.)